src/terms/entropy.jl

```diff
@@ -5,26 +5,20 @@ This is in particular useful because the free energy,
 not the energy, is minimized at self-consistency.
 """
 struct Entropy end
-(::Entropy)(basis) = TermEntropy(basis)
+(::Entropy)(::AbstractBasis) = TermEntropy()
```

Comment:
why?

Reply:
You rather like `::Any`?

Reply:
I just mean `basis` is more clear than `::AbstractBasis` to me

Reply:
but that's an unused variable.

Reply:
which is bad why?

Reply:
Arguably no real difference here, but still flags up by my static code analysis.

Reply:
The answer to that is fix the tools, not fix the code... (which is why I'm very distrustful of analysis/formatting tools). Unused variables are completely fine, esp. with julia's somewhat informal way of doing interfaces. Anyway, doesn't really matter